### DISCUSSION TITLE
treating python version as a dependency in pyproject.toml using incorrect lookups

### DISCUSSION THREAD

**Discussion by ITProKyle** [renovatebot/renovate#24256] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.81.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

With #24236 now parsing and handling python as a dependency (using docker as the datasource) I am now receiving the following error/warning (redacted some private info):

```text
 WARN: Excess registryUrls found for datasource lookup - using first configured only (repository=example-repo)
       "datasource": "docker",
       "packageName": "python",
       "registryUrls": [
         "https://nexus.foobar.com/repository/pypi-releases/simple",
         "https://pypi.org/pypi"
       ]
```

My company uses nexus for all of our python dependencies so I use the following `pyproject.toml`:

```toml
[[tool.poetry.source]]
name = "pypi-releases"
url = "https://nexus.foobar.com/repository/pypi-releases/simple"
priority = "explicit"
```

Looking through my renovate config, i don't have anything overriding `registryUrls` for python so my assumption is that this could only come from a combination of the default pypi datasource registry url and the contents of my `pyproject.toml` file.

To me, this appears to be a bug in that the registry urls extracted for the pypi datasource are being used for the docker datasource.

Aside from this, is there an easy way to disable managing the python version? I know this could likely be done with `packageRules` but an option on the manager itself would be nice. e.g.

```json
{
    "poetry": {
        "managePythonVersion": false
    }
}
```

> **NOTE:** A reproduction repository has not been provided as this is using a private repository but I would assume one could be created if needed using [test.pypi.org](https://test.pypi.org/) as the other source.

### Relevant debug logs

<details><summary>Logs</summary>

```
 WARN: Excess registryUrls found for datasource lookup - using first configured only (repository=example-repo)
       "datasource": "docker",
       "packageName": "python",
       "registryUrls": [
         "https://nexus.foobar.com/repository/pypi-releases/simple",
         "https://pypi.org/pypi"
       ]
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
I think we'll roll this back until we understand better why it went wrong

**Reply by rarkins** [reply, marked as answer]
You should be able to disable this with matchPackageNames=["python"], matchManagers=["poetry"], enabled=false

**Reply by ITProKyle** [reply]
Perfect, thanks!

**Reply by ITProKyle** [reply]
Small update on this, adding the following to the `config.js` of my bot did not work (added it to this file for testing from my local machine or I would have added it to my default preset):

```js
module.exports = {
  ...,
  packageRules: [
    {
      description: "temporary workaround for https://github.com/renovatebot/renovate/pull/24257",
      enabled: false,
      matchManagers: ["poetry"],
      matchPackageNames: ["python"]
    }
  ],
}
```

I had to wrap it in `force` (which I'm not the biggest fan of using) for it to take effect:

```js
module.exports = {
  ...,
  force: {
    packageRules: [
      {
        description: "temporary workaround for https://github.com/renovatebot/renovate/pull/24257",
        enabled: false,
        matchManagers: ["poetry"],
        matchPackageNames: ["python"]
      }
    ]
  },
}
```
